extension/lib/shopify/collection-repository/command/ListGraphQl.js

```diff
@@ -0,0 +1,89 @@
+const DEFAULT_NUM_PER_PAGE = 250
+
+class ShopifyCollectionRepositoryCommandListGraphQl {
+  /**
+   * @param {ShopifyStorefrontClient} storefrontClient
+   */
+  constructor (storefrontClient) {
+    this._storefrontClient = storefrontClient
+    this._numberPerPage = DEFAULT_NUM_PER_PAGE
+  }
+
+  /**
+   * @returns {Promise<ShopifyCollectionRepositoryCommandGetOutput[]>}
+   */
+  async execute () {
+    const regExp = new RegExp(/([0-9])\w+/)
+    const collectionsEdges = await this._makePaginatedRequests()
+    return collectionsEdges.map(edge => {
+      let image = null
+      if (edge.node.image && edge.node.image.originalSrc) {
+        image = edge.node.image.originalSrc
+      }
+      return {
+        id: regExp.exec(Buffer.from(edge.node.id.toString(), 'base64').toString())[0],
+        title: edge.node.title,
+        handle: edge.node.handle,
+        image: image
+      }
+    })
+  }
+
+  /**
+   * @param {string?} cursor
+   * @return {Promise<Object[]>}
+   * @private
+   */
+  async _makePaginatedRequests (cursor) {
```

Comment:
I would rename the method to something like: fetchCollections

Reply:
done